integrations/transportation/ryanair.js

```diff
@@ -0,0 +1,135 @@
+  }
+
+  return {
+    token: res.body.token,
+    customerId: res.body.customerId,
+  };
+}
+
+async function connect(requestLogin, requestWebView) {
+  // Here we can request credentials etc..
+
+  // Here we can use two functions to invoke screens
+  // requestLogin() or requestWebView()
+  const { username, password } = await requestLogin();
+
+  if (!(password || '').length) {
+    throw Error('Password cannot be empty');
+  }
+
+  return logIn(username, password);
+}
+
+function disconnect() {
+  // Here we should do any cleanup (deleting tokens etc..)
+  return {};
+}
+
+async function getAllFlights(entries, customerId, token) {
+  const allFlights = [];
+
+  await Promise.all(entries.map(async (entry) => {
+    await agent
+      .put(`${PROFILE_URL}${customerId}/bookings/booking`)
+      .set('Accept', 'application/json')
+      .set('X-Auth-Token', token)
+      .send({
+        bookingId: entry.bookingId,
+        pnr: entry.pnr,
+      })
+      .then((res) => {
+        if (res.ok) {
+          res.body.Flights.forEach((singleFlight) => {
+            allFlights.push({ 
+              bookingId: res.body.BookingId,
+              flightInfo: singleFlight,
+              pnr: res.body.RecordLocator,
+            });
+          });
+        }
+      });
+  }));
+  return allFlights;
+}
+
+async function collect(state) {
+  const { token, customerId } = state;
+
+  const pastBookings = await agent
+    .get(`${PROFILE_URL}${customerId}/bookings/past`)
+    .set('Accept', 'application/json')
+    .set('X-Auth-Token', token);
+
+  if (!pastBookings.ok) {
+    console.error('--------RESPONSE---------', pastBookings);
+    throw Error('Error while fetching past bookings.');
+  }
+
+  const entries = pastBookings.body.bookings.filter(entry => entry.status === 'Confirmed');
+
+  // WHY: there can be multiple flights in one booking
+  const activities = await getAllFlights(entries, customerId, token)
+    .then((allFlights) => {
+      // WHY: there can be multiple bookings for the same flight (e.g. for different passengers)
+      // HOW: filters flights with the same flight number
+      const uniqueFlights = Array.from(new Set(allFlights.map(a => a.flightInfo.FlightNumber)))
+        .map(mappedFlightNumber => allFlights.find(a => a.flightInfo.FlightNumber === mappedFlightNumber));
+
+      return Object.values(uniqueFlights)
+        .map(k => ({
+          // TODO: make sure it's always unique
+          id: `B${k.bookingId}${k.flightInfo.FlightNumber}PNR${k.pnr}`,
+          datetime: k.flightInfo.DepartLocal,
+          // TODO: calculate distance between airports using some API
+          // distanceKilometers: ,
+          durationHours: moment(k.flightInfo.Arrive).diff(moment(k.flightInfo.Depart), 'minutes') / 60,
+          activityType: ACTIVITY_TYPE_TRANSPORTATION,
+          transportationMode: TRANSPORTATION_MODE_PLANE,
+          carrier: 'Ryanair',
+          // TODO: "translate" city name abrrievations to full names
```

Comment:
maybe the abbreviation is an airportcode?
If that's the case, then you should populate `departureAirportCode` and `destinationAirportCode` instead of station.